### DISCUSSION TITLE
another late night rambling: propelling ShapeShift dex to the next level

### DISCUSSION THREAD

**Discussion by LPX55** [shapeshift/web#1311] [General]
Much of the world's _successful and/or industry leading_ FOSS projects have similar traits that distinguishes them from the rest: plugins, themes, customizability, resulting in a huge dev community. Let's take a prime example: WordPress. Regardless of what your opinions may be regarding PHP or WP itself, the matter of the fact is that it continues to be the platform of choice for more than half of the world's websites. We all make fun of WP, the numbers speak for themselves. The abundance of plugins, themes, and tools to develop a site that suits specific needs of bloggers, businesses, and the everything in between. At the same time, there's also a very clear reason why WordPress powered-sites are the most commonly hacked/defaced sites. In a senior research project during my senior year in college (almost a decade ago... fuck, we're getting old), nearly 64% of WP-powered websites in my county in Southern California were found to be either exploitable or already exploited --  simply because the SMB/SMEs never bothered to update outdated software. Most small to medium business hired  webdevs to create the site, then proceeded to neglect it without ever upgrading plugins, themes, or even the worldpress even as zero-day vulns were popping up left and right.

But no one can deny that WordPress has made an enormous impact on the growth of the web from 2012 to 2020. In fact, its arguable that WP ranks in the top 5 most influential FOSS platforms of web2. And consequently, an enormous secondary market of plugins and themes was born, and continues to be a 8-figure industry to this day. 

Its my personal opinion these similar concepts from web2 will be an important factor in the success of web3 projects. AFAIK, ShapeShift is the only project atm that is truly trying to be a FOSS that aims to be chain-agnostic, is easily deployable locally, and has a community mindset of innovation and 'do whatever you want and contribute to the project as long as it adds value.'

Obviously comparing WP and ShapeShift platform is kinda like comparing oranges and apples, but is it really *that different*? Yes, security is absolutely paramount in an app that deals with assets and finance. But let's be real here: the abundance of DeFi projects and DEXes is something that can not be ignored. Without a truly game-changing model or idea, we will be fighting an uphill battle trying to compete with the new heavyweights of DeFi. 

Although I am not directly in the eng. workstream, for months I've been battling with myself with this question: is what we're doing 'enough' to regain ShapeShift's influence and spot as a top exchange in crypto? Since we're in FOSS mode, should we be more focused on building a platform that really resonates with the traditional open source community of pluggability and customizability? The app store changed the game for smartphones, just as any it has for any platform that allowed everyone and anyone to build what *they want* on top of the existing OS/framework/software. 

Take a look around the DeFi scene -- its mostly just forks of other projects, just on a different L2 or slightly modified. IMO the true FOSS spirit was never and isn't meant to be just a bunch of forks deployed on another network. Where's the innovation? Or at least experimentation with new UI/UX flows? 

When WordPress was first conceptualized and launched as a blogging platform, no one would have guessed that it would quickly take over the web as the #1 CMS. What if ShapeShift takes the success stories of the widely adopted web2 foss projects and take inspiration. DeFi by definition should be decentralized, but it feels as if we're trying too hard to keep the users on the 'official' ShapeShift app -- which is not a bad thing, it makes sense from a business standpoint. But WordPress, Laravel, Vue, React, Chrome, VLC, Android, etc. etc. didn't become leaders in their field for having superior tech -- they are what they are because of the enormous dev community that innovated on top of the respective frameworks and softwares. 

The ability for users to add their own custom plugins/modules to the platform undoubtedly brings risks. But just as you wouldn't install a random chrome extension without doing a little bit of research, I have faith that as the DeFi world matures, so will the end-users' ability to discern a a suspicious plugin/module from a legitimate one that adds value on top to their experience. The secondary marketplace and industry that could potentially grow out of this approach to the platform, in my opinion, would far outweigh the the handful of users that mindlessly install every plugin they see,  As the user numbers grow, naturally the dev community would follow and grow. Both popular and non-popular/new DeFi platforms would be compelled to integrate their services on top of the ShapeShift platform, without the DAO having to dish out bounties. Give them the platform and users, they will come on their own. 

Imagine if ShapeShift platform became the <insert any FOSS project with huge third party dev/enterprise communities> pumping out plugins that can be integrated seamlessly on top of ShapeShift platform. The resulting ripple effect would be huge, and ShapeShift would once again be regarded as an organization pushing the boundaries of the status quo.

This would be a huge undertaking with major refactoring of the existing alpha platform, and I am not even sure if this is possible. But I am curious on what people think, and maybe start a discussion on how the platform can one day become a self-evolving and self-improving project thriving with developers, value adders, and take back the throne that ShapeShift had held for many years.

Remember:

![](https://gateway.pinata.cloud/ipfs/QmawSNNdJDQR6UPbkZcJr7ePEoMexpA9kL5oGoMuxhHqwQ)

**Comment by LPX55** [comment]
looks like we've got Discussions unlocked! https://github.com/shapeshift/web/discussions

closing issue

**Reply by mrnerdhair** [reply]
i found a button to transfer issues to discussions!

**Reply by LPX55** [reply]
Glad my late night soju inspired ramblings won't be buried to never be seen again 😅
